vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -0,0 +1,135 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from pyVmomi import vim
+
+from datadog_checks.base import ensure_unicode
+from datadog_checks.vsphere.constants import MOR_TYPE_AS_STRING, SHORT_ROLLUP
+
+
+def format_metric_name(counter):
+    return "{}.{}.{}".format(
+        ensure_unicode(counter.groupInfo.key),
+        ensure_unicode(counter.nameInfo.key),
+        ensure_unicode(SHORT_ROLLUP[str(counter.rollupType)]),
+    )
+
+
+def match_any_regex(string, regexes):
+    for regex in regexes:
+        match = regex.match(string)  # FIXME: Should we use re.IGNORECASE like legacy?
+        if match:
+            return True
+    return False
+
+
+def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
+    resource_type = MOR_TYPE_AS_STRING[type(mor)]
+
+    if not [f for f in resource_filters if f[0] == resource_type]:
+        # No filter for this resource, collect everything
+        return False
```

Comment:
```python
    for f in resource_filters:
        if f[0] == resource_type:
            break
    else:
        return False
```

Reply:
🤔 How strongly do you prefer this? I think I'd rather have a one-liner for this as the logic is separate from what the the method is doing.

It's much easier to understand like this I think especially with the comment and because nothing else relies on this.

thoughts?

Reply:
I was just trying to not iterate through everything, up to you

Reply:
That's fair but I think we can ignore the performance impact of this :)